fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -959,15 +959,18 @@ private void createComplexRecords(int count,
         }
     }
 
-    private static LoggableTimeoutException findTimeoutException(final RecordCoreException e) {
+    private static LoggableKeysAndValues<? extends Exception> findTimeoutException(final RecordCoreException e) {
         Map<Throwable, String> visited = new IdentityHashMap<>();
         ArrayDeque<Throwable> toVisit = new ArrayDeque<>();
         toVisit.push(e);
         while (!toVisit.isEmpty()) {
             Throwable cause = toVisit.removeFirst();
             if (!visited.containsKey(cause)) {
+                // TODO: This should eventually be removed since LoggableTimeoutException would not be thrown
```

Comment:
Why not remove now?

Reply:
This would still be thrown when the property is set to use the legacy asyncToSync. Changed the comment.